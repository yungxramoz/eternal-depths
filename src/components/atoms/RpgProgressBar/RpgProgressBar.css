.rpgui-progress{
    margin: 0;
}