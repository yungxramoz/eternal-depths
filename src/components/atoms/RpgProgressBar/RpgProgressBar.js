import React from 'react'
import './RpgProgressBar.css'

const RpgProgressBar = ({ max, current, type }) => {
  const progress = (current / max) * 100
  let color = 'red'
  switch (type) {
    case 'hp':
      color = 'red'
      break
    case 'xp':
      color = 'purple'
      break
    default:
      break
  }

  return (
    <div className="rpgui-progress" data-rpguitype="progress">
      <div className=" rpgui-progress-track">
        <div className={`rpgui-progress-fill ${color}`} style={{left: '0px', width: `${progress}%`}}></div>
      </div>
      <div className=" rpgui-progress-left-edge"></div>
      <div className=" rpgui-progress-right-edge"></div>
    </div>
  )
}

export default RpgProgressBar
