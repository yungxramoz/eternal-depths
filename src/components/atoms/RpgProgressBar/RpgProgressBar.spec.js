import { render, screen } from '@testing-library/react'
import React from 'react'
import RpgProgressBar from './RpgProgressBar'

describe('RpgProgressBar', () => {
  it('renders correctly', () => {
    render(<RpgProgressBar max={100} current={100} type="hp" />)
    expect(document.body).toMatchSnapshot('Progress bar at 100%')
  })

  it('calculates progress correctly', () => {
    render(<RpgProgressBar max={100} current={50} />)
    expect(document.body).toMatchSnapshot('Progress bar at 50%')
  })

  it('sets color correctly based on type', () => {
    render(<RpgProgressBar max={100} current={50} type="hp" />)
    expect(document.body).toMatchSnapshot('HP progress bar')

    render(<RpgProgressBar max={100} current={50} type="xp" />)
    expect(document.body).toMatchSnapshot('XP progress bar')
  })
})
