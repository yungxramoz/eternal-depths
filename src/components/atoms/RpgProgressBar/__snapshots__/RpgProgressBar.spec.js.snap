// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpgProgressBar calculates progress correctly: Progress bar at 50% 1`] = `
<body>
  <div>
    <div
      class="rpgui-progress"
      data-rpguitype="progress"
    >
      <div
        class=" rpgui-progress-track"
      >
        <div
          class="rpgui-progress-fill red progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
      <div
        class=" rpgui-progress-left-edge"
      />
      <div
        class=" rpgui-progress-right-edge"
      />
    </div>
  </div>
</body>
`;

exports[`RpgProgressBar renders correctly: Progress bar at 100% 1`] = `
<body>
  <div>
    <div
      class="rpgui-progress"
      data-rpguitype="progress"
    >
      <div
        class=" rpgui-progress-track"
      >
        <div
          class="rpgui-progress-fill red progress-animation"
          style="left: 0px; width: 100%;"
        />
      </div>
      <div
        class=" rpgui-progress-left-edge"
      />
      <div
        class=" rpgui-progress-right-edge"
      />
    </div>
  </div>
</body>
`;

exports[`RpgProgressBar sets color correctly based on type: HP progress bar 1`] = `
<body>
  <div>
    <div
      class="rpgui-progress"
      data-rpguitype="progress"
    >
      <div
        class=" rpgui-progress-track"
      >
        <div
          class="rpgui-progress-fill red progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
      <div
        class=" rpgui-progress-left-edge"
      />
      <div
        class=" rpgui-progress-right-edge"
      />
    </div>
  </div>
</body>
`;

exports[`RpgProgressBar sets color correctly based on type: XP progress bar 1`] = `
<body>
  <div>
    <div
      class="rpgui-progress"
      data-rpguitype="progress"
    >
      <div
        class=" rpgui-progress-track"
      >
        <div
          class="rpgui-progress-fill red progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
      <div
        class=" rpgui-progress-left-edge"
      />
      <div
        class=" rpgui-progress-right-edge"
      />
    </div>
  </div>
  <div>
    <div
      class="rpgui-progress"
      data-rpguitype="progress"
    >
      <div
        class=" rpgui-progress-track"
      >
        <div
          class="rpgui-progress-fill purple progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
      <div
        class=" rpgui-progress-left-edge"
      />
      <div
        class=" rpgui-progress-right-edge"
      />
    </div>
  </div>
</body>
`;
